#!/usr/bin/env python3
PACKAGE = "follow_lane_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add("thresh", int_t, 0, "Thresh Value", 180, 0, 255)
gen.add("speed", double_t, 0, "Speed Value", 1.5, 0.0, 4.0)
gen.add("enable_drive", bool_t, 0, "Enable", False)

gen.add("canny_thresh_l",int_t,0,"Lower thresh for canny line finding",700,0,100000)
gen.add("canny_thresh_u", int_t, 0, "Upper thresh for canny line finding", 2000,0,100000)

gen.add("blur_kernal", int_t, 0, "Median blur kernal", 9, 3, 15)

gen.add("dilation_base", int_t, 0, "Dilation amount", 4, 1, 10)

gen.add("minimum_slope", double_t, 0, "Min slope for Hough lines", 1,0.1,10)
gen.add("maximum_slope", double_t, 0, "Max slope for Hough lines", 3,0.1,10)
gen.add("lines_rho", int_t, 0 , "Rho for lines", 1,1,10)
gen.add("lines_thresh", int_t, 0 , "Lines Threshold", 100, 1, 1000)
gen.add("lines_min_len", int_t, 0, "Minimum Line Length",100,1,1000)

gen.add("slopes_thresh",int_t,0,"Lower Threshold for a Vertical",10,1,25)

gen.add("turn_speed_const",double_t,0,"Bigger = turn harder",10,0.1,1000)
gen.add("turn_max",double_t,0,"Most the car can turn while turning",3,0.1,100)
gen.add("turn_min",double_t,0,"Least the car can turn while turning",0.5,0.1,100)

exit(gen.generate(PACKAGE, "follow_lane", "FollowLaneHough"))
